webob/compat.py

```diff
@@ -197,6 +198,15 @@ def read_multi(self, environ, keep_blank_values, strict_parsing):
                     self.bytes_read += len(hdr_text)
                     parser.feed(hdr_text.decode(self.encoding, self.errors))
                     headers = parser.close()
+                    # Some clients add Content-Length for part headers, ignore them
+                    if 'content-length' in headers:
```

Comment:
+1

I like this better than just throwing out the Content-Length data entirely. I can't help but feel like `FieldStorage` needs to be fixed so that it does the right thing even if the `Content-Length` exists vs just throwing it out. I think using the fact that something has `Content-Length` as a reason to use a binary tempfile is a bad idea.